File: zig/easy/complement-of-base-10-integer.zig
const std = @import("std");

pub fn bitwiseComplement(n: u64) u64 {
    if (n == 0) return 1;

    const bit_len = std.math.log2_int(u64, n) + 1;
    const mask: u64 = (@as(u64, 1) << bit_len) - 1;

    return (~n) & mask;
}

File: c/easy/complement-of-base-10-integer.c
int bitwiseComplement(int n) {
    if (n == 0) return 1;

    int bit_length = 0;
    int temp = n;

    while (temp) {
        bit_length++;
        temp >>= 1;
    }

    int mask = (1 << bit_length) - 1;
    return (~n) & mask;
}
